# Data Acquisition
* This file is being used to find an appropriate data source
* Initial exploration of data from multiple sources

# Imports

In [138]:
import requests
import numpy as np
import pandas as pd
import seaborn as sns
import pandas as pd
from xml.etree import ElementTree

import matplotlib.pyplot as plt
# Only works inside notebook
%matplotlib inline 
#import matplotlib as mpl
#mpl.rcParams['agg.path.chunksize'] = 10000 # assists with processor speed

# import preprocessing
from sklearn.preprocessing import MinMaxScaler 
from sklearn.preprocessing import StandardScaler
from sklearn.preprocessing import RobustScaler
from sklearn.preprocessing import QuantileTransformer

from sklearn.neighbors import KNeighborsClassifier

# import helper files from local environment
from env import user, password, host
import QMCBT_00_quicktips as qt
import QMCBT_01_acquire as acquire
import QMCBT_02_prepare as prepare
import QMCBT_03_explore as explore
import QMCBT_04_model as model
import QMCBT_05_evaluate as evaluate
import QMCBT_explore_evaluate as ee
import QMCBT_wrangle as w

# allows import reload without needing to clear kernel and rerun
from importlib import reload
# reload(packagename) 

import warnings
warnings.filterwarnings("ignore")

**CUSTOM EXPLORATION FUNCTIONS
nunique_column_all(df): PRINT NUNIQUE OF ALL COLUMNS
nunique_column_objects(df): PRINT NUNIQUE OF COLUMNS THAT ARE OBJECTS
nunique_column_qty(df): PRINT NUNIQUE OF COLUMNS THAT ARE *NOT* OBJECTS
numeric_range(df): COMPUTE RANGE FOR ALL NUMERIC VARIABLES

**USEFUL EXPLORATORY CODE**
DFNAME.head()
DFNAME.shape
DFNAME.shape[0] #read row count
DFNAME.describe().T
DFNAME.columns.to_list()
DFNAME.COLUMNNAME.value_counts(dropna=False)
DFNAME.dtypes
DFNAME.select_dtypes(include='object').columns
DFNAME.select_dtypes(include='float').columns
pd.crosstab(DFNAME.COLUMN-1, DFNAME.COLUMN-2)


In [139]:
# Left Align Tables in Jupyter Notebook
from IPython.core.display import HTML
table_css = 'table {align:left;display:block}'
HTML('<style>{}</style>'.format(table_css))

<div class="alert alert-success">

# TX Hate Crimes (by Victim Type)
* https://crime-data-explorer.fr.cloud.gov/pages/docApi
* https://api.usa.gov/crime/fbi/sapi/api/hatecrime/states/TX/all/VICTIM_TYPE?API_KEY=iiHnOKfno2Mgkt5AynpvPpUQTEyxE77jo1RU8PIv

In [37]:
url = 'https://api.usa.gov/crime/fbi/sapi/api/hatecrime/states/TX/all/VICTIM_TYPE?API_KEY=iiHnOKfno2Mgkt5AynpvPpUQTEyxE77jo1RU8PIv'
response = requests.get(url)
response

<Response [200]>

In [38]:
response.ok

True

In [39]:
response.text

'{\n  "ui_type" : "basic_table",\n  "noun" : "Victim",\n  "category" : "Victim demographic",\n  "title" : "Victim Type",\n  "short_title" : "Victim Type",\n  "ui_restriction" : null,\n  "keys" : [ "Other", "Business", "Religious Organization", "Financial Institution", "Society/Public", "Individual", "Unknown", "Government", "Law Enforcement Officer" ],\n  "data" : [ {\n    "value" : 26,\n    "data_year" : 1993,\n    "month_num" : 0,\n    "key" : "Other"\n  }, {\n    "value" : 0,\n    "data_year" : 2011,\n    "month_num" : 0,\n    "key" : "Other"\n  }, {\n    "value" : 0,\n    "data_year" : 2010,\n    "month_num" : 0,\n    "key" : "Other"\n  }, {\n    "value" : 0,\n    "data_year" : 2002,\n    "month_num" : 0,\n    "key" : "Other"\n  }, {\n    "value" : 1,\n    "data_year" : 2000,\n    "month_num" : 0,\n    "key" : "Other"\n  }, {\n    "value" : 0,\n    "data_year" : 2008,\n    "month_num" : 0,\n    "key" : "Other"\n  }, {\n    "value" : 0,\n    "data_year" : 2001,\n    "month_num" : 0,

In [40]:
data = response.json()
print(type(data))
data

<class 'dict'>


{'ui_type': 'basic_table',
 'noun': 'Victim',
 'category': 'Victim demographic',
 'title': 'Victim Type',
 'short_title': 'Victim Type',
 'ui_restriction': None,
 'keys': ['Other',
  'Business',
  'Religious Organization',
  'Financial Institution',
  'Society/Public',
  'Individual',
  'Unknown',
  'Government',
  'Law Enforcement Officer'],
 'data': [{'value': 26, 'data_year': 1993, 'month_num': 0, 'key': 'Other'},
  {'value': 0, 'data_year': 2011, 'month_num': 0, 'key': 'Other'},
  {'value': 0, 'data_year': 2010, 'month_num': 0, 'key': 'Other'},
  {'value': 0, 'data_year': 2002, 'month_num': 0, 'key': 'Other'},
  {'value': 1, 'data_year': 2000, 'month_num': 0, 'key': 'Other'},
  {'value': 0, 'data_year': 2008, 'month_num': 0, 'key': 'Other'},
  {'value': 0, 'data_year': 2001, 'month_num': 0, 'key': 'Other'},
  {'value': 1, 'data_year': 2009, 'month_num': 0, 'key': 'Other'},
  {'value': 0, 'data_year': 2012, 'month_num': 0, 'key': 'Other'},
  {'value': 11, 'data_year': 1998, 'month_n

In [41]:
response = requests.get(url)
print(response)
response.json()

<Response [200]>


{'ui_type': 'basic_table',
 'noun': 'Victim',
 'category': 'Victim demographic',
 'title': 'Victim Type',
 'short_title': 'Victim Type',
 'ui_restriction': None,
 'keys': ['Other',
  'Business',
  'Religious Organization',
  'Financial Institution',
  'Society/Public',
  'Individual',
  'Unknown',
  'Government',
  'Law Enforcement Officer'],
 'data': [{'value': 26, 'data_year': 1993, 'month_num': 0, 'key': 'Other'},
  {'value': 0, 'data_year': 2011, 'month_num': 0, 'key': 'Other'},
  {'value': 0, 'data_year': 2010, 'month_num': 0, 'key': 'Other'},
  {'value': 0, 'data_year': 2002, 'month_num': 0, 'key': 'Other'},
  {'value': 1, 'data_year': 2000, 'month_num': 0, 'key': 'Other'},
  {'value': 0, 'data_year': 2008, 'month_num': 0, 'key': 'Other'},
  {'value': 0, 'data_year': 2001, 'month_num': 0, 'key': 'Other'},
  {'value': 1, 'data_year': 2009, 'month_num': 0, 'key': 'Other'},
  {'value': 0, 'data_year': 2012, 'month_num': 0, 'key': 'Other'},
  {'value': 11, 'data_year': 1998, 'month_n

In [42]:
response = requests.get(url)
data = response.json()
data.keys()

dict_keys(['ui_type', 'noun', 'category', 'title', 'short_title', 'ui_restriction', 'keys', 'data', 'precise_data'])

In [44]:
# Pull it all at once
response = requests.get(url)
data = response.json()

TX_hate_crimes_victim_df = pd.DataFrame(data['data'])
TX_hate_crimes_victim_df

value  data_year  month_num                      key
0       26       1993          0                    Other
1        0       2011          0                    Other
2        0       2010          0                    Other
3        0       2002          0                    Other
4        1       2000          0                    Other
..     ...        ...        ...                      ...
238      0       1992          0  Law Enforcement Officer
239      0       1994          0  Law Enforcement Officer
240      0       1996          0  Law Enforcement Officer
241      0       2016          0  Law Enforcement Officer
242      0       1991          0  Law Enforcement Officer

[243 rows x 4 columns]

In [45]:
TX_hate_crimes_victim_df.key.unique()

array(['Other', 'Business', 'Religious Organization',
       'Financial Institution', 'Society/Public', 'Individual', 'Unknown',
       'Government', 'Law Enforcement Officer'], dtype=object)

<div class="alert alert-success">

# TX Hate Crimes (by location)
* https://crime-data-explorer.fr.cloud.gov/pages/docApi
* https://api.usa.gov/crime/fbi/sapi/api/hatecrime/states/TX/all/LOCATION?API_KEY=iiHnOKfno2Mgkt5AynpvPpUQTEyxE77jo1RU8PIv

In [2]:
url = 'https://api.usa.gov/crime/fbi/sapi/api/hatecrime/states/TX/all/LOCATION?API_KEY=iiHnOKfno2Mgkt5AynpvPpUQTEyxE77jo1RU8PIv'
response = requests.get(url)
response

<Response [200]>

In [3]:
response.ok

True

In [5]:
response.text

'{\n  "ui_type" : "basic_table",\n  "noun" : "offense",\n  "category" : "Location characteristics",\n  "title" : "Location Type",\n  "short_title" : null,\n  "ui_restriction" : null,\n  "keys" : [ "Residence Home", "Parking Garage/Lot", "Abandoned Condemned/Structure", "Air/Bus/Train Terminal", "Amusement Park", "Arena/Stadium/Fairgrounds", "ATM Separate From Bank", "Auto Dealership", "Bank", "Bar/Nightclub", "Campground", "Church/Synagogue/Temple/Mosque", "Commercial/Office Building", "Community Center", "Construction Site", "Cyberspace", "Daycare Facility", "Department/Discount Store", "Dock/Wharf/Shipping Terminal", "Drug Store/Doctors Office/Hospital", "Farm Facility", "Field/Woods", "Gambling Facility/Casino/Race Track", "Government/Public Building", "Grocery Store", "Highway/Alley/Street/Sidewalk", "Hotel/Motel", "Industrial Site", "Jail/Prison/Corrections Facility", "Lake/Waterway/Beach", "Liquor Store", "Military Base", "Unknown", "Park/Playground", "Rental Storage Facility", "

In [6]:
data = response.json()
print(type(data))
data

<class 'dict'>


{'ui_type': 'basic_table',
 'noun': 'offense',
 'category': 'Location characteristics',
 'title': 'Location Type',
 'short_title': None,
 'ui_restriction': None,
 'keys': ['Residence Home',
  'Parking Garage/Lot',
  'Abandoned Condemned/Structure',
  'Air/Bus/Train Terminal',
  'Amusement Park',
  'Arena/Stadium/Fairgrounds',
  'ATM Separate From Bank',
  'Auto Dealership',
  'Bank',
  'Bar/Nightclub',
  'Campground',
  'Church/Synagogue/Temple/Mosque',
  'Commercial/Office Building',
  'Community Center',
  'Construction Site',
  'Cyberspace',
  'Daycare Facility',
  'Department/Discount Store',
  'Dock/Wharf/Shipping Terminal',
  'Drug Store/Doctors Office/Hospital',
  'Farm Facility',
  'Field/Woods',
  'Gambling Facility/Casino/Race Track',
  'Government/Public Building',
  'Grocery Store',
  'Highway/Alley/Street/Sidewalk',
  'Hotel/Motel',
  'Industrial Site',
  'Jail/Prison/Corrections Facility',
  'Lake/Waterway/Beach',
  'Liquor Store',
  'Military Base',
  'Unknown',
  'Park/

In [7]:
response = requests.get(url)
print(response)
response.json()

<Response [200]>


{'ui_type': 'basic_table',
 'noun': 'offense',
 'category': 'Location characteristics',
 'title': 'Location Type',
 'short_title': None,
 'ui_restriction': None,
 'keys': ['Residence Home',
  'Parking Garage/Lot',
  'Abandoned Condemned/Structure',
  'Air/Bus/Train Terminal',
  'Amusement Park',
  'Arena/Stadium/Fairgrounds',
  'ATM Separate From Bank',
  'Auto Dealership',
  'Bank',
  'Bar/Nightclub',
  'Campground',
  'Church/Synagogue/Temple/Mosque',
  'Commercial/Office Building',
  'Community Center',
  'Construction Site',
  'Cyberspace',
  'Daycare Facility',
  'Department/Discount Store',
  'Dock/Wharf/Shipping Terminal',
  'Drug Store/Doctors Office/Hospital',
  'Farm Facility',
  'Field/Woods',
  'Gambling Facility/Casino/Race Track',
  'Government/Public Building',
  'Grocery Store',
  'Highway/Alley/Street/Sidewalk',
  'Hotel/Motel',
  'Industrial Site',
  'Jail/Prison/Corrections Facility',
  'Lake/Waterway/Beach',
  'Liquor Store',
  'Military Base',
  'Unknown',
  'Park/

In [8]:
response = requests.get(url)
data = response.json()
data.keys()

dict_keys(['ui_type', 'noun', 'category', 'title', 'short_title', 'ui_restriction', 'keys', 'data', 'precise_data'])

In [9]:
ui_type = data['ui_type']
noun = data['noun']
category = data['category']
title = data['title']
short_title = data['short_title']
ui_restriction = data['ui_restriction']
api_keys_column = data['keys']
api_data_column = data['data']


print(f'ui_type: {ui_type}')
print(f'noun: {noun}')
print(f'category: {category}')
print(f'title: {title}')
print(f'short_title: {short_title}')
print(f'ui_restriction: {ui_restriction}')
print(f'api_keys_column: {api_keys_column}')
print(f'api_data_column: {api_data_column}')


ui_type: basic_table
noun: offense
category: Location characteristics
title: Location Type
short_title: None
ui_restriction: None
api_keys_column: ['Residence Home', 'Parking Garage/Lot', 'Abandoned Condemned/Structure', 'Air/Bus/Train Terminal', 'Amusement Park', 'Arena/Stadium/Fairgrounds', 'ATM Separate From Bank', 'Auto Dealership', 'Bank', 'Bar/Nightclub', 'Campground', 'Church/Synagogue/Temple/Mosque', 'Commercial/Office Building', 'Community Center', 'Construction Site', 'Cyberspace', 'Daycare Facility', 'Department/Discount Store', 'Dock/Wharf/Shipping Terminal', 'Drug Store/Doctors Office/Hospital', 'Farm Facility', 'Field/Woods', 'Gambling Facility/Casino/Race Track', 'Government/Public Building', 'Grocery Store', 'Highway/Alley/Street/Sidewalk', 'Hotel/Motel', 'Industrial Site', 'Jail/Prison/Corrections Facility', 'Lake/Waterway/Beach', 'Liquor Store', 'Military Base', 'Unknown', 'Park/Playground', 'Rental Storage Facility', 'Rest Area', 'Restaurant', 'School/College', 'Scho

In [12]:
import pandas as pd
df = pd.DataFrame(data['data'])
df

value  data_year  month_num                key
0       165       1993          0     Residence Home
1        63       2010          0     Residence Home
2       136       2002          0     Residence Home
3       144       2001          0     Residence Home
4        65       2009          0     Residence Home
...     ...        ...        ...                ...
1237      5       2008          0  Convenience Store
1238      3       1991          0  Convenience Store
1239     11       1992          0  Convenience Store
1240      6       2004          0  Convenience Store
1241      8       1996          0  Convenience Store

[1242 rows x 4 columns]

In [13]:
df.month_num.unique()

array([0])

In [14]:
df.data_year.unique()

array([1993, 2010, 2002, 2001, 2009, 2012, 2000, 1998, 2014, 2015, 2017,
       1997, 2005, 2007, 2003, 2011, 2006, 1995, 1999, 2013, 1994, 2016,
       2008, 1991, 1992, 2004, 1996])

In [15]:
df.data_year.value_counts()

1993    46
2003    46
2004    46
1992    46
1991    46
2008    46
2016    46
1994    46
2013    46
1999    46
1995    46
2006    46
2011    46
2007    46
2010    46
2005    46
1997    46
2017    46
2015    46
2014    46
1998    46
2000    46
2012    46
2009    46
2001    46
2002    46
1996    46
Name: data_year, dtype: int64

In [16]:
df.value.unique()

array([165,  63, 136, 144,  65,  67,  99, 106,  51,  66, 129, 103, 104,
       118,  75,  91, 120,  47, 148,  79, 102,  28, 173, 111, 137,  34,
        18,  57,  60,  25,  23,  38,  29,  17,  26,  41,  31,   8,  20,
        32,  14,  42,  36,  45,   0,   1,   4,   2,   3,   7,   6,   9,
         5,  10,  19,  11,  16,  13,  12, 119,  73,  82,  24,  78,  58,
        62,  27,  55,  81,  64, 105, 115,  59,  74,  15,  22,  21])

In [18]:
df.isnull().sum()

value        0
data_year    0
month_num    0
key          0
dtype: int64

In [17]:
# Pull it all at once
response = requests.get(url)
data = response.json()

TX_hate_crimes_df = pd.DataFrame(data['data'])
TX_hate_crimes_df

value  data_year  month_num                key
0       165       1993          0     Residence Home
1        63       2010          0     Residence Home
2       136       2002          0     Residence Home
3       144       2001          0     Residence Home
4        65       2009          0     Residence Home
...     ...        ...        ...                ...
1237      5       2008          0  Convenience Store
1238      3       1991          0  Convenience Store
1239     11       1992          0  Convenience Store
1240      6       2004          0  Convenience Store
1241      8       1996          0  Convenience Store

[1242 rows x 4 columns]

<div class="alert alert-success">

# National Hate Crimes (by Location)
* https://crime-data-explorer.fr.cloud.gov/pages/docApi
* https://api.usa.gov/crime/fbi/sapi/api/hatecrime/national/all/LOCATION?API_KEY=iiHnOKfno2Mgkt5AynpvPpUQTEyxE77jo1RU8PIv

In [28]:
url = 'https://api.usa.gov/crime/fbi/sapi/api/hatecrime/national/all/LOCATION?API_KEY=iiHnOKfno2Mgkt5AynpvPpUQTEyxE77jo1RU8PIv'

In [29]:
response = requests.get(url)
response

<Response [200]>

In [30]:
response.ok

True

In [31]:
response.status_code

200

In [32]:
response.text

'{\n  "ui_type" : "basic_table",\n  "noun" : "offense",\n  "category" : "Location characteristics",\n  "title" : "Location Type",\n  "short_title" : null,\n  "ui_restriction" : null,\n  "keys" : [ "Residence Home", "Parking Garage/Lot", "Abandoned Condemned/Structure", "Air/Bus/Train Terminal", "Amusement Park", "Arena/Stadium/Fairgrounds", "ATM Separate From Bank", "Auto Dealership", "Bank", "Bar/Nightclub", "Campground", "Church/Synagogue/Temple/Mosque", "Commercial/Office Building", "Community Center", "Construction Site", "Cyberspace", "Daycare Facility", "Department/Discount Store", "Dock/Wharf/Shipping Terminal", "Drug Store/Doctors Office/Hospital", "Farm Facility", "Field/Woods", "Gambling Facility/Casino/Race Track", "Government/Public Building", "Grocery Store", "Highway/Alley/Street/Sidewalk", "Hotel/Motel", "Industrial Site", "Jail/Prison/Corrections Facility", "Lake/Waterway/Beach", "Liquor Store", "Military Base", "Unknown", "Park/Playground", "Rental Storage Facility", "

In [33]:
data = response.json()
print(type(data))
data

<class 'dict'>


{'ui_type': 'basic_table',
 'noun': 'offense',
 'category': 'Location characteristics',
 'title': 'Location Type',
 'short_title': None,
 'ui_restriction': None,
 'keys': ['Residence Home',
  'Parking Garage/Lot',
  'Abandoned Condemned/Structure',
  'Air/Bus/Train Terminal',
  'Amusement Park',
  'Arena/Stadium/Fairgrounds',
  'ATM Separate From Bank',
  'Auto Dealership',
  'Bank',
  'Bar/Nightclub',
  'Campground',
  'Church/Synagogue/Temple/Mosque',
  'Commercial/Office Building',
  'Community Center',
  'Construction Site',
  'Cyberspace',
  'Daycare Facility',
  'Department/Discount Store',
  'Dock/Wharf/Shipping Terminal',
  'Drug Store/Doctors Office/Hospital',
  'Farm Facility',
  'Field/Woods',
  'Gambling Facility/Casino/Race Track',
  'Government/Public Building',
  'Grocery Store',
  'Highway/Alley/Street/Sidewalk',
  'Hotel/Motel',
  'Industrial Site',
  'Jail/Prison/Corrections Facility',
  'Lake/Waterway/Beach',
  'Liquor Store',
  'Military Base',
  'Unknown',
  'Park/

In [34]:
response = requests.get(url)
print(response)
response.json()

<Response [200]>


{'ui_type': 'basic_table',
 'noun': 'offense',
 'category': 'Location characteristics',
 'title': 'Location Type',
 'short_title': None,
 'ui_restriction': None,
 'keys': ['Residence Home',
  'Parking Garage/Lot',
  'Abandoned Condemned/Structure',
  'Air/Bus/Train Terminal',
  'Amusement Park',
  'Arena/Stadium/Fairgrounds',
  'ATM Separate From Bank',
  'Auto Dealership',
  'Bank',
  'Bar/Nightclub',
  'Campground',
  'Church/Synagogue/Temple/Mosque',
  'Commercial/Office Building',
  'Community Center',
  'Construction Site',
  'Cyberspace',
  'Daycare Facility',
  'Department/Discount Store',
  'Dock/Wharf/Shipping Terminal',
  'Drug Store/Doctors Office/Hospital',
  'Farm Facility',
  'Field/Woods',
  'Gambling Facility/Casino/Race Track',
  'Government/Public Building',
  'Grocery Store',
  'Highway/Alley/Street/Sidewalk',
  'Hotel/Motel',
  'Industrial Site',
  'Jail/Prison/Corrections Facility',
  'Lake/Waterway/Beach',
  'Liquor Store',
  'Military Base',
  'Unknown',
  'Park/

In [35]:
response = requests.get(url)

data = response.json()
data.keys()

dict_keys(['ui_type', 'noun', 'category', 'title', 'short_title', 'ui_restriction', 'keys', 'data', 'precise_data'])

In [36]:
# Pull it all at once
response = requests.get(url)
data = response.json()

national_hate_crimes_df = pd.DataFrame(data['data'])
national_hate_crimes_df

value  data_year  month_num                key
0      1915       2015          0     Residence Home
1      2374       1999          0     Residence Home
2      2679       2008          0     Residence Home
3      2315       2005          0     Residence Home
4      2909       1996          0     Residence Home
...     ...        ...        ...                ...
1237    103       2003          0  Convenience Store
1238     67       2009          0  Convenience Store
1239     94       2006          0  Convenience Store
1240     58       1991          0  Convenience Store
1241     92       1992          0  Convenience Store

[1242 rows x 4 columns]

<div class="alert alert-success">

# Life Expectancy
* **GHO Selections:** https://apps.who.int/gho/athena/api/GHO
* **Homepage:** https://www.who.int/data/gho/info/athena-api-examples
* **CSV Data Pull:** https://apps.who.int/gho/athena/api/GHO/WHOSIS_000001,WHOSIS_000002,WHOSIS_000003,WHOSIS_000004,WHOSIS_000005,WHOSIS_000006,WHOSIS_000007,WHOSIS_000008,WHOSIS_000009,WHOSIS_000010,WHOSIS_000012,WHOSIS_000013,WHOSIS_000014,WHOSIS_000015,WHOSIS_000016?format=csv

In [101]:
life_Expectancy_df = pd.read_csv('../my_notes/smoking_WHOSIS000001-16.csv')

/var/folders/x8/x12zllv954v785j2tdhggqrr0000gn/T/ipykernel_87080/2383627457.py:1: DtypeWarning: Columns (1,8) have mixed types. Specify dtype option on import or set low_memory=False.
  life_Expectancy_df = pd.read_csv('../my_notes/smoking_WHOSIS000001-16.csv')


In [102]:
life_Expectancy_df

GHO      DATASOURCE PUBLISHSTATE  YEAR REGION  \
0      WHOSIS_000004             NaN    PUBLISHED  2000    AFR   
1      WHOSIS_000004             NaN    PUBLISHED  2001    AFR   
2      WHOSIS_000004             NaN    PUBLISHED  2003    AFR   
3      WHOSIS_000004             NaN    PUBLISHED  2004    AFR   
4      WHOSIS_000004             NaN    PUBLISHED  2006    AFR   
...              ...             ...          ...   ...    ...   
39875  WHOSIS_000003             NaN    PUBLISHED  2008    AFR   
39876  WHOSIS_000003             NaN    PUBLISHED  2015    AFR   
39877  WHOSIS_000003             NaN    PUBLISHED  2017    AFR   
39878  WHOSIS_000006  NUT_NGA2018DHS    PUBLISHED  2018    AFR   
39879  WHOSIS_000006  NUT_PAK2018NNS    PUBLISHED  2018    EMR   

      WORLDBANKINCOMEGROUP COUNTRY   SEX UNSDGREGION        Display Value  \
0                      NaN     NaN  FMLE         NaN                  371   
1                      NaN     NaN   MLE         NaN                  422   
2                      NaN     NaN  FMLE         NaN                  379   
3                      NaN     NaN   MLE         NaN                  417   
4                      NaN     NaN  FMLE         NaN                  356   
...                    ...     ...   ...         ...                  ...   
39875                  NaN     ZWE  BTSX         NaN   31.19 [27.96-34.7]   
39876                  NaN     ZWE  BTSX         NaN  27.76 [23.64-32.67]   
39877                  NaN     ZWE  BTSX         NaN   27.01 [21.78-33.8]   
39878               WB_LMI     NGA   NaN         NaN     28.7 [26.8-30.6]   
39879               WB_LMI     PAK   NaN         NaN     47.8 [46.2-49.5]   

         Numeric       Low      High  StdErr  StdDev  \
0      371.42360       NaN       NaN     NaN     NaN   
1      421.53050       NaN       NaN     NaN     NaN   
2      378.79310       NaN       NaN     NaN     NaN   
3      416.63960       NaN       NaN     NaN     NaN   
4      356.15590       NaN       NaN     NaN     NaN   
...          ...       ...       ...     ...     ...   
39875   31.19215  27.96338  34.70319     NaN     NaN   
39876   27.76433  23.63723  32.66603     NaN     NaN   
39877   27.01455  21.77675  33.79541     NaN     NaN   
39878   28.68000  26.80000  30.60000     NaN     NaN   
39879   47.84000  46.20000  49.50000     NaN     NaN   

                                                Comments  
0                                                    NaN  
1                                                    NaN  
2                                                    NaN  
3                                                    NaN  
4                                                    NaN  
...                                                  ...  
39875                                                NaN  
39876                                                NaN  
39877                                                NaN  
39878  Nigeria Demographic and Health Survey 2018. Ab...  
39879            Pakistan 2018 National Nutrition Survey  

[39880 rows x 16 columns]

In [103]:
life_Expectancy_df.isnull().sum()

GHO                         0
DATASOURCE              39874
PUBLISHSTATE                0
YEAR                        0
REGION                    646
WORLDBANKINCOMEGROUP    39019
COUNTRY                  1780
SEX                      4968
UNSDGREGION             39460
Display Value               0
Numeric                     0
Low                     19229
High                    19229
StdErr                  39880
StdDev                  39880
Comments                39213
dtype: int64

In [104]:
life_Expectancy_df.YEAR.unique()

array([2000, 2001, 2003, 2004, 2006, 2007, 2008, 2009, 2010, 2011, 2012,
       2014, 2015, 2002, 2005, 2013, 2016, 2018, 2017, 2019, 1990, 1996,
       1987, 1999, 1993, 1995, 1988, 1994, 1989, 1997, 2020, 1992, 1998,
       1991, 1986, 1979, 1981, 1971, 1978, 1980, 1985, 1977, 1984, 1975,
       1972, 1974, 1983, 1956, 1958, 1965, 1967, 1976, 1970, 1961, 1968,
       1964, 1966, 1973, 1982, 1960, 1962, 1963, 1969, 1959, 1955, 1952,
       1954, 1957, 1953, 1951])

In [105]:
life_Expectancy_df.GHO.unique()

array(['WHOSIS_000004', 'WHOSIS_000014', 'WHOSIS_000002', 'WHOSIS_000007',
       'WHOSIS_000015', 'WHOSIS_000001', 'WHOSIS_000006', 'WHOSIS_000016',
       'WHOSIS_000003'], dtype=object)

In [106]:
life_Expectancy_df.GHO.isnull().sum()

0

In [107]:
life_Expectancy_df.GHO.value_counts()

WHOSIS_000004    9690
WHOSIS_000003    9648
WHOSIS_000016    6262
WHOSIS_000014    4353
WHOSIS_000002    2328
WHOSIS_000007    2328
WHOSIS_000015    2328
WHOSIS_000001    2328
WHOSIS_000006     615
Name: GHO, dtype: int64

In [136]:
life_Expectancy_df.shape[0]

39880

In [137]:
print(f'WHOSIS_000004    {round(9690/39880*100,2)}%')
print(f'WHOSIS_000003    {round(9648/39880*100,2)}%')
print(f'WHOSIS_000016     {round(6262/39880*100,2)}%')
print(f'WHOSIS_000014     {round(4353/39880*100,2)}%')
print(f'WHOSIS_000002     {round(2328/39880*100,2)}%')
print(f'WHOSIS_000007     {round(2328/39880*100,2)}%')
print(f'WHOSIS_000015     {round(2328/39880*100,2)}%')
print(f'WHOSIS_000001     {round(2328/39880*100,2)}%')
print(f'WHOSIS_000006     {round(615/39880*100,2)}%')

WHOSIS_000004    24.3%
WHOSIS_000003    24.19%
WHOSIS_000016     15.7%
WHOSIS_000014     10.92%
WHOSIS_000002     5.84%
WHOSIS_000007     5.84%
WHOSIS_000015     5.84%
WHOSIS_000001     5.84%
WHOSIS_000006     1.54%


<div class="alert alert-info">

#### About 24% of data
* 'WHOSIS_000004': 'Adult mortality rate (probability of dying between 15 and 60 years per 1000 population)',
* 'WHOSIS_000003': 'Neonatal mortality rate (0 to 27 days) per 1000 live births)',

#### Between 10-15% of data
* 'WHOSIS_000016': 'Mortality rate among children ages 5 to 9 years (per 1000 children aged 5)', 
* 'WHOSIS_000014': 'Stillbirth rate (per 1000 total births)', 

#### These are all exactly the same record count at 5.84%
* 'WHOSIS_000001': 'Life expectancy at birth (years)', 
* 'WHOSIS_000002': 'Healthy life expectancy (HALE) at birth (years)', 
* 'WHOSIS_000007': 'Healthy life expectancy (HALE) at age 60 (years)', 
* 'WHOSIS_000015': 'Life expectancy at age 60 (years)', 

#### Very little representation at only 1.54% with just 615 records
* 'WHOSIS_000006': 'Infants exclusively breastfed for the first six months of life (%)', 

'WHOSIS_000001': 'Life expectancy at birth (years)', 
'WHOSIS_000002': 'Healthy life expectancy (HALE) at birth (years)', 
'WHOSIS_000003': 'Neonatal mortality rate (0 to 27 days) per 1000 live births) (SDG 3.2.2)',
'WHOSIS_000004': 'Adult mortality rate (probability of dying between 15 and 60 years per 1000 population)',
'WHOSIS_000006': 'Infants exclusively breastfed for the first six months of life (%)', 
'WHOSIS_000007': 'Healthy life expectancy (HALE) at age 60 (years)', 
'WHOSIS_000014': 'Stillbirth rate (per 1000 total births)', 
'WHOSIS_000015': 'Life expectancy at age 60 (years)', 
'WHOSIS_000016': 'Mortality rate among children ages 5 to 9 years (per 1000 children aged 5)', 

|[GHO Code](https://apps.who.int/gho/athena/api/GHO)      |[Documentation](https://www.who.int/data/gho/indicator-metadata-registry)|[Global Health Observatory](https://www.who.int/data/gho) (GHO) Code Description|
|:-------------|:-----------:|:------------------------------------------------|
|WHOSIS_000001|[📖](https://www.who.int/data/gho/indicator-metadata-registry/imr-details/65)|Life expectancy at birth (years)| 
|WHOSIS_000002|[📖](https://www.who.int/data/gho/indicator-metadata-registry/imr-details/66)|Healthy life expectancy (HALE) at birth (years)| 
|WHOSIS_000003|[📖](https://www.who.int/data/gho/indicator-metadata-registry/imr-details/67)|Neonatal mortality rate (0 to 27 days) per 1000 live births)|
|WHOSIS_000004|[📖](https://www.who.int/data/gho/indicator-metadata-registry/imr-details/64)|Adult mortality rate (probability of dying between 15 and 60 years per 1000 population)|
|WHOSIS_000006|[📖](https://www.who.int/data/gho/indicator-metadata-registry/imr-details/130)|Infants exclusively breastfed for the first six months of life (%)| 
|WHOSIS_000007|[📖](https://www.who.int/data/gho/indicator-metadata-registry/imr-details/3443)|Healthy life expectancy (HALE) at age 60 (years)| 
|WHOSIS_000014|[📖](https://www.who.int/data/gho/indicator-metadata-registry/imr-details/2444)|Stillbirth rate (per 1000 total births)| 
|WHOSIS_000015|[📖](https://www.who.int/data/gho/indicator-metadata-registry/imr-details/2977)|Life expectancy at age 60 (years)| 
|WHOSIS_000016|[📖](https://www.who.int/data/gho/indicator-metadata-registry/imr-details/5669)|Mortality rate among children ages 5 to 9 years (per 1000 children aged 5)| 


In [108]:
life_Expectancy_df.PUBLISHSTATE.unique()

array(['PUBLISHED'], dtype=object)

In [109]:
life_Expectancy_df.Numeric.unique()

array([371.4236 , 421.5305 , 378.7931 , ...,  27.01455,  28.68   ,
        47.84   ])

In [110]:
life_Expectancy_df.COUNTRY.nunique()

195

In [111]:
life_Expectancy_df.COUNTRY.unique()

array([nan, 'AFG', 'AGO', 'ALB', 'ARE', 'ARG', 'ARM', 'ATG', 'AUS', 'AUT',
       'AZE', 'BDI', 'BEL', 'BEN', 'BFA', 'BGD', 'BGR', 'BHR', 'BHS',
       'BIH', 'BLR', 'BLZ', 'BOL', 'BRA', 'BRB', 'BRN', 'BTN', 'BWA',
       'CAF', 'CAN', 'CHE', 'CHL', 'CHN', 'CIV', 'CMR', 'COD', 'COG',
       'COL', 'COM', 'CPV', 'CRI', 'CUB', 'CYP', 'CZE', 'DEU', 'DJI',
       'DNK', 'DOM', 'DZA', 'ECU', 'EGY', 'ERI', 'ESP', 'EST', 'ETH',
       'FIN', 'FJI', 'FRA', 'FSM', 'GAB', 'GBR', 'GEO', 'GHA', 'GIN',
       'GMB', 'GNB', 'GNQ', 'GRC', 'GRD', 'GTM', 'GUY', 'HND', 'HRV',
       'HTI', 'HUN', 'IDN', 'IND', 'IRL', 'IRN', 'IRQ', 'ISL', 'ISR',
       'ITA', 'JAM', 'JOR', 'JPN', 'KAZ', 'KEN', 'KGZ', 'KHM', 'KIR',
       'KOR', 'KWT', 'LAO', 'LBN', 'LBR', 'LBY', 'LCA', 'LKA', 'LSO',
       'LTU', 'LUX', 'LVA', 'MAR', 'MDA', 'MDG', 'MDV', 'MEX', 'MKD',
       'MLI', 'MLT', 'MMR', 'MNE', 'MNG', 'MOZ', 'MRT', 'MUS', 'MWI',
       'MYS', 'NAM', 'NER', 'NGA', 'NIC', 'NLD', 'NOR', 'NPL', 'NZL',
       'OMN', '

In [112]:
life_Expectancy_df.COUNTRY.isnull().sum()

1780

In [113]:
life_Expectancy_df.COUNTRY.isnull().sum() / smoking_df.shape[0]

0.044633901705115345

In [114]:
life_Expectancy_df.SEX.isnull().sum()

4968

In [115]:
life_Expectancy_df.SEX.isnull().sum() / smoking_df.shape[0]

0.12457372116349047

In [77]:
life_expectancy_df = smoking_df[(smoking_df['GHO'] == 'WHOSIS_000001') |
                                (smoking_df['GHO'] == 'WHOSIS_000002') |
                                (smoking_df['GHO'] == 'WHOSIS_000007') |
                                (smoking_df['GHO'] == 'WHOSIS_000015')
                               ]

In [78]:
life_expectancy_df

GHO DATASOURCE PUBLISHSTATE  YEAR  REGION  \
13331  WHOSIS_000002        NaN    PUBLISHED  2000     EMR   
13332  WHOSIS_000007        NaN    PUBLISHED  2000     EMR   
13333  WHOSIS_000015        NaN    PUBLISHED  2015     EMR   
13334  WHOSIS_000001        NaN    PUBLISHED  2019     EMR   
13335  WHOSIS_000007        NaN    PUBLISHED  2000     EMR   
...              ...        ...          ...   ...     ...   
24246  WHOSIS_000015        NaN    PUBLISHED  2019  GLOBAL   
24247  WHOSIS_000001        NaN    PUBLISHED  2000  GLOBAL   
24248  WHOSIS_000015        NaN    PUBLISHED  2000  GLOBAL   
24249  WHOSIS_000007        NaN    PUBLISHED  2010  GLOBAL   
24250  WHOSIS_000002        NaN    PUBLISHED  2015  GLOBAL   

      WORLDBANKINCOMEGROUP COUNTRY   SEX UNSDGREGION Display Value   Numeric  \
13331                  NaN     AFG   MLE         NaN          46.9  46.93113   
13332                  NaN     AFG   MLE         NaN          10.2  10.22541   
13333                  NaN     AFG   MLE         NaN          15.7  15.67539   
13334                  NaN     AFG   MLE         NaN          63.3  63.28709   
13335                  NaN     AFG  FMLE         NaN          10.1  10.11293   
...                    ...     ...   ...         ...           ...       ...   
24246      WB_WORLD_INCOME     NaN  FMLE         NaN          22.7  22.67449   
24247      WB_WORLD_INCOME     NaN  BTSX         NaN          66.8  66.79430   
24248      WB_WORLD_INCOME     NaN  BTSX         NaN          18.8  18.83003   
24249      WB_WORLD_INCOME     NaN  BTSX         NaN          15.1  15.08183   
24250      WB_WORLD_INCOME     NaN  BTSX         NaN          62.9  62.93812   

       Low  High  StdErr  StdDev Comments  
13331  NaN   NaN     NaN     NaN      NaN  
13332  NaN   NaN     NaN     NaN      NaN  
13333  NaN   NaN     NaN     NaN      NaN  
13334  NaN   NaN     NaN     NaN      NaN  
13335  NaN   NaN     NaN     NaN      NaN  
...    ...   ...     ...     ...      ...  
24246  NaN   NaN     NaN     NaN      NaN  
24247  NaN   NaN     NaN     NaN      NaN  
24248  NaN   NaN     NaN     NaN      NaN  
24249  NaN   NaN     NaN     NaN      NaN  
24250  NaN   NaN     NaN     NaN      NaN  

[9312 rows x 16 columns]

In [79]:
life_expectancy_df.GHO.unique()

array(['WHOSIS_000002', 'WHOSIS_000007', 'WHOSIS_000015', 'WHOSIS_000001'],
      dtype=object)

<div class="alert alert-success">

# Infant Mortality
* **GHO Selections:** https://apps.who.int/gho/athena/api/GHO
* **Homepage:** https://www.who.int/data/gho/info/athena-api-examples
* **CSV Data Pull:** https://apps.who.int/gho/athena/api/GHO/MDG_0000000001,MDG_0000000003,MDG_0000000005,MDG_0000000007,MDG_0000000025,MDG_0000000026?format=csv

In [99]:
infant_mortality_df = pd.read_csv('../my_notes/infant_mortality.csv')

/var/folders/x8/x12zllv954v785j2tdhggqrr0000gn/T/ipykernel_87080/2740071651.py:1: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.
  infant_mortality_df = pd.read_csv('../my_notes/infant_mortality.csv')


In [100]:
infant_mortality_df

GHO PUBLISHSTATE       YEAR  REGION  UNREGION COUNTRY  \
0      MDG_0000000026    PUBLISHED       2004     AFR       NaN     SLE   
1      MDG_0000000026    PUBLISHED       2000     AFR       NaN     SSD   
2      MDG_0000000026    PUBLISHED       2005     AFR       NaN     SLE   
3      MDG_0000000026    PUBLISHED       2001     AFR       NaN     SSD   
4      MDG_0000000026    PUBLISHED       2002     AFR       NaN     CAF   
...               ...          ...        ...     ...       ...     ...   
82930  MDG_0000000003    PUBLISHED  2010-2015     EMR       NaN     NaN   
82931  MDG_0000000003    PUBLISHED  2000-2005     WPR       NaN     NaN   
82932  MDG_0000000003    PUBLISHED  2010-2015     WPR       NaN     NaN   
82933  MDG_0000000025    PUBLISHED  2001-2007  GLOBAL       NaN     NaN   
82934  MDG_0000000025    PUBLISHED  2015-2021  GLOBAL       NaN     NaN   

         AGEGROUP   SEX            Display Value     Numeric     Low    High  \
0             NaN   NaN  1 850 [ 1 480 - 2 330 ]  1850.00000  1480.0  2330.0   
1             NaN   NaN  1 730 [ 1 320 - 2 250 ]  1730.00000  1320.0  2250.0   
2             NaN   NaN  1 760 [ 1 410 - 2 220 ]  1760.00000  1410.0  2220.0   
3             NaN   NaN  1 690 [ 1 310 - 2 190 ]  1690.00000  1310.0  2190.0   
4             NaN   NaN    1 280 [ 872 - 1 880 ]  1280.00000   872.0  1880.0   
...           ...   ...                      ...         ...     ...     ...   
82930  YEARS15-19  FMLE                     50.4    50.38059     NaN     NaN   
82931  YEARS15-19  FMLE                     13.4    13.35170     NaN     NaN   
82932  YEARS15-19  FMLE                     14.4    14.37810     NaN     NaN   
82933         NaN   NaN                     64.1    64.12033     NaN     NaN   
82934         NaN   NaN                     83.6    83.56604     NaN     NaN   

       StdErr  StdDev                                           Comments  
0         NaN     NaN                                                NaN  
1         NaN     NaN                                                NaN  
2         NaN     NaN                                                NaN  
3         NaN     NaN                                                NaN  
4         NaN     NaN                                                NaN  
...       ...     ...                                                ...  
82930     NaN     NaN  World population prospects: 2019 revision. Uni...  
82931     NaN     NaN  World population prospects: 2019 revision. Uni...  
82932     NaN     NaN  World population prospects: 2019 revision. Uni...  
82933     NaN     NaN  UNICEF/WHO joint global database on Skilled at...  
82934     NaN     NaN  UNICEF/WHO joint global database on Skilled at...  

[82935 rows x 15 columns]

In [116]:
infant_mortality_df.isnull().sum()

GHO                  0
PUBLISHSTATE         0
YEAR                 0
REGION             154
UNREGION         82781
COUNTRY           1631
AGEGROUP         77590
SEX               5573
Display Value        0
Numeric              0
Low               7498
High              7498
StdErr           82935
StdDev           82935
Comments         75437
dtype: int64

In [117]:
infant_mortality_df.YEAR.unique()

array([2004, 2000, 2005, 2001, 2002, 2003, 2010, 2007, 2011, 2013, 2012,
       2015, 2009, 2014, 2016, 2006, 2008, 2017, 1990, 1991, 1993, 1994,
       1996, 1997, 1998, 1999, 2018, 2019, 1992, 1995, 2020, 1955, 1956,
       1957, 1958, 1959, 1961, 1962, 1964, 1965, 1967, 1970, 1973, 1975,
       1976, 1978, 1979, 1981, 1982, 1983, 1984, 1986, 1987, 1989, 1951,
       1952, 1954, 1960, 1963, 1966, 1968, 1969, 1971, 1972, 1974, 1977,
       1980, 1985, 1988, 1953, 1950, 1949, 1940, 1942, 1947, 1946, 1948,
       1945, 1943, 1944, 1935, 1936, 1937, 1938, 1941, 1939, 1934, 1933,
       1932, '1998', '2000', '2001', '2003', '2004', '2006', '2007',
       '2008', '2009', '2011', '2012', '2014', '2015', '2017', '2020',
       '1953', '1955', '1956', '1958', '1959', '1961', '1962', '1963',
       '1964', '1966', '1967', '1969', '1970', '1972', '1975', '1978',
       '1980', '1981', '1983', '1984', '1986', '1987', '1988', '1989',
       '1990', '1991', '1992', '1994', '1995', '1997', '2016', 

In [118]:
infant_mortality_df.GHO.unique()

array(['MDG_0000000026', 'MDG_0000000001', 'MDG_0000000007',
       'MDG_0000000003', 'MDG_0000000025'], dtype=object)

In [119]:
infant_mortality_df.GHO.isnull().sum()

0

In [120]:
infant_mortality_df.GHO.value_counts()

MDG_0000000007    36084
MDG_0000000001    35933
MDG_0000000003     5345
MDG_0000000026     3420
MDG_0000000025     2153
Name: GHO, dtype: int64

In [129]:
infant_mortality_df.shape[0]

82935

In [135]:
print(f'MDG_0000000007    {round(36084/82935*100,2)}%')
print(f'MDG_0000000001    {round(35933/82935*100,2)}%')
print(f'MDG_0000000003     {round(5345/82935*100,2)}%')
print(f'MDG_0000000026     {round(3420/82935*100,2)}%')
print(f'MDG_0000000025     {round(2153/82935*100,2)}%')

MDG_0000000007    43.51%
MDG_0000000001    43.33%
MDG_0000000003     6.44%
MDG_0000000026     4.12%
MDG_0000000025     2.6%


<div class="alert alert-info">

#### Over 35K value counts
* 'MDG_0000000007': 'Under-five mortality rate (per 1000 live births)', 
* 'MDG_0000000001': 'Infant mortality rate (between birth and 11 months per 1000 live births)', 

#### Between 2-5k value counts
* 'MDG_0000000003': 'Adolescent birth rate (per 1000 women)', 
* 'MDG_0000000026': 'Maternal mortality ratio (per 100 000 live births)', 
* 'MDG_0000000025': 'Births attended by skilled health personnel (%)', 

#### MDG...05 returned no data
* 'MDG_0000000005': 'Contraceptive prevalence (%)', 

'MDG_0000000001': 'Infant mortality rate (between birth and 11 months per 1000 live births)', 
'MDG_0000000003': 'Adolescent birth rate (per 1000 women)', 
'MDG_0000000005': 'Contraceptive prevalence (%)', 
'MDG_0000000007': 'Under-five mortality rate (per 1000 live births)', 
'MDG_0000000025': 'Births attended by skilled health personnel (%)', 
'MDG_0000000026': 'Maternal mortality ratio (per 100 000 live births)', 

|[GHO Code](https://apps.who.int/gho/athena/api/GHO)      |[Documentation](https://www.who.int/data/gho/indicator-metadata-registry)|[Global Health Observatory](https://www.who.int/data/gho) (GHO) Code Description|
|:-------------|:-----------:|:------------------------------------------------|
|MDG_0000000001|[📖](https://www.who.int/data/gho/indicator-metadata-registry/imr-details/1)|Infant mortality rate (between birth and 11 months per 1000 live births)| 
|MDG_0000000003|[📖](https://www.who.int/data/gho/indicator-metadata-registry/imr-details/4669)|Adolescent birth rate (per 1000 women)| 
|MDG_0000000005|[📖](https://www.who.int/data/gho/indicator-metadata-registry/imr-details/5)|Contraceptive prevalence (%)| 
|MDG_0000000007|[📖](https://www.who.int/data/gho/indicator-metadata-registry/imr-details/7)|Under-five mortality rate (per 1000 live births)| 
|MDG_0000000025|[📖](https://www.who.int/data/gho/indicator-metadata-registry/imr-details/25)|Births attended by skilled health personnel (%)| 
|MDG_0000000026|[📖](https://www.who.int/data/gho/indicator-metadata-registry/imr-details/26)|Maternal mortality ratio (per 100 000 live births)| 

In [121]:
infant_mortality_df.PUBLISHSTATE.unique()

array(['PUBLISHED'], dtype=object)

In [122]:
infant_mortality_df.Numeric.unique()

array([1850.     , 1730.     , 1760.     , ...,   14.3781 ,   64.12033,
         83.56604])

In [123]:
infant_mortality_df.COUNTRY.nunique()

195

In [124]:
infant_mortality_df.COUNTRY.unique()

array(['SLE', 'SSD', 'CAF', 'SOM', 'AFG', 'TCD', 'NGA', 'GNB', 'RWA',
       'GIN', 'MRT', 'LBR', 'GMB', 'ETH', 'TLS', 'NER', 'TZA', 'MLI',
       'CIV', 'BDI', 'DJI', 'KEN', 'COG', 'LSO', 'ERI', 'MWI', 'SWZ',
       'COD', 'CMR', 'ZWE', 'HTI', 'MDG', 'SEN', 'NPL', 'LAO', 'BFA',
       'ZMB', 'TGO', 'AGO', 'BEN', 'SDN', 'COM', 'UGA', 'KHM', 'BGD',
       'MOZ', 'GHA', 'GNQ', 'GAB', 'PNG', 'BTN', 'NAM', 'IND', 'PAK',
       'BOL', 'SLB', 'YEM', 'IDN', 'MMR', 'BWA', 'FSM', 'PRK', 'STP',
       'VUT', 'GUY', 'MAR', 'DZA', 'ZAF', 'NIC', 'PHL', 'KIR', 'SUR',
       'IRQ', 'VEN', 'GTM', 'LCA', 'WSM', 'LBY', 'PER', 'TON', 'MNG',
       'CPV', 'ECU', 'BHS', 'SYC', 'PAN', 'VCT', 'COL', 'KGZ', 'PRY',
       'MUS', 'MDV', 'JAM', 'HND', 'TTO', 'DOM', 'SLV', 'VNM', 'BLZ',
       'ARG', 'BRA', 'ATG', 'JOR', 'ROU', 'LKA', 'RUS', 'KAZ', 'TUN',
       'FJI', 'BRB', 'MEX', 'EGY', 'CUB', 'GEO', 'THA', 'ARM', 'AZE',
       'SYR', 'CHN', 'MDA', 'UZB', 'BRN', 'TJK', 'CRI', 'MYS', 'LVA',
       'GRD', 'UKR',

In [125]:
infant_mortality_df.COUNTRY.isnull().sum()

1631

In [126]:
infant_mortality_df.COUNTRY.isnull().sum() / smoking_df.shape[0]

0.04089769307923771

In [127]:
infant_mortality_df.SEX.isnull().sum()

5573

In [128]:
infant_mortality_df.SEX.isnull().sum() / smoking_df.shape[0]

0.13974423269809427

<div class="alert alert-success">

# Checkpoint #1

1. a brief (one-sentence) description of your project
* Predict future Life Expectancy or Healthy Life Expectancy

2. a link to the data source
* **GHO Selections:** https://apps.who.int/gho/athena/api/GHO
* **Homepage:** https://www.who.int/data/gho/info/athena-api-examples
* **CSV Data Pull:** https://apps.who.int/gho/athena/api/GHO/WHOSIS_000001,WHOSIS_000002,WHOSIS_000003,WHOSIS_000004,WHOSIS_000005,WHOSIS_000006,WHOSIS_000007,WHOSIS_000008,WHOSIS_000009,WHOSIS_000010,WHOSIS_000012,WHOSIS_000013,WHOSIS_000014,WHOSIS_000015,WHOSIS_000016?format=csv 

3. your target variable
* Numeric (Expected Years of continued Life)

4. what one observation represents
* Numeric for one year in specified country according to category of life expectancy
    * 'WHOSIS_000001': 'Life expectancy at birth (years)', 
    * 'WHOSIS_000002': 'Healthy life expectancy (HALE) at birth (years)', 
    * 'WHOSIS_000007': 'Healthy life expectancy (HALE) at age 60 (years)', 
    * 'WHOSIS_000015': 'Life expectancy at age 60 (years)', 


<div class="alert alert-danger">

# Sometimes it is faster to read the unreadable data than it is to read about how to make the data readable!